### DISCUSSION TITLE
RemoveParamRector

### DISCUSSION THREAD

**Discussion by brandonkelly** [rectorphp/rector#7043] [Ideas]
We have a couple methods in Craft 4 that have dropped an argument, and it would be great if there was a way to automatically remove any code calling them via Rector. For example:

```php
$services->set(RemoveParamRector::class)
    ->configure([
        new RemoveParam('MyClass', 'methodName', 2),
    ]);
```

Which would cause the following change:

```diff
 /** @var MyClass $obj */
-$obj->methodName($arg1, $arg2, $arg3, $arg4);
+$obj->methodName($arg1, $arg2, $arg4);
```

**Comment by TomasVotruba** [comment]
Added in https://github.com/rectorphp/rector-src/pull/1906/files

**Reply by brandonkelly** [reply]
This doesn’t appear to be working in my test – am I doing something wrong? https://github.com/craftcms/rector/commit/db915054d524161f2138d0e286f9487b3e6df0cf

**Reply by TomasVotruba** [reply]
Ah, it seem it's static call. This rule handles method call, based on the example :smile:

**Reply by brandonkelly** [reply]
Ah whoops, I need both, but didn’t think about how the implementation would be different.

**Reply by TomasVotruba** [reply]
No troubles :+1:  
In php-parser, mostly any expression is different node:

* https://github.com/rectorphp/php-parser-nodes-docs#phpparsernodeexprmethodcall
* https://github.com/rectorphp/php-parser-nodes-docs#phpparsernodeexprstaticcall

I've added `StaticCall` support here:  https://github.com/rectorphp/rector-src/pull/1916/files

**Reply by brandonkelly** [reply]
Amazing! Works like a charm now. Thanks so much!

**Reply by TomasVotruba** [reply]
You're welcome 👍
